### PR TITLE
Calculate batch metrics & support metric aggregation

### PR DESCRIPTION
**Pull request by sankalp04**

### CODE REVIEW

File: batch_metrics.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import logging
+import argparse
+import pandas as pd
+from app.settings import settings
+from datatron.governor import MetricsManager, MetricsAggregator
+
+logging.basicConfig(format=settings.DEFAULT_LOG_FORMAT, level=logging.INFO)
+
+# Change this to a mounted folder path & add this to settings potentially
+METRICS_DIR = "/tmp/metrics/"
```
Comment (dhruvrrp) [inline batch_metrics.py:14]: Lets switch this to a volume mount, which will allow you to share the metadata between jobs

File: batch_metrics.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    @log_time_taken(f"Calculating batch metrics for prediction file: {self.prediction_filepath} for job-id: {self.job_id}")
+    def process_batch(self):
+        chunksize = 1e6 # Make this a variable in the settings or determine dynamically
+        try:
+            for prediction_chunk in pd.read_csv(self.prediction_file, chunksize=chunksize):
+                for feedback_file in self.feedback_files:
+                    for feedback_chunk in pd.read_csv(feedback_file, chunksize=chunksize):
+                        try:
+                            joined_df = pd.merge(prediction_chunk, feedback_chunk, left_on = "datatron_request_id", right_on = "feedback_id", how="inner")
+                            self.metrics_manager.update_batch(joined_df["feedback"].to_numpy(), joined_df["predictions"].to_numpy()) # VERIFY COLUMN NAMES
```
Comment (dhruvrrp) [inline batch_metrics.py:76]: isnt this batch_update in governor

File: batch_metrics.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            with open(file_name, "w") as fopen:
+                json.dump(metric_vals, fopen)
+        except FileNotFoundError as e:
+            logging.info(f"Metrics for matadata calculated for job-id {job_id} could not be saved due to file {file_name} not being found.")
+
+    @log_time_taken(f"Calculating batch metrics for prediction file: {self.prediction_filepath} for job-id: {self.job_id}")
+    def process_batch(self):
+        chunksize = 1e6 # Make this a variable in the settings or determine dynamically
+        try:
+            for prediction_chunk in pd.read_csv(self.prediction_file, chunksize=chunksize):
```
Comment (dhruvrrp) [inline batch_metrics.py:71]: Don't the prediction and feedback files need to be copied over?

**Review by hpierce1312 — APPROVED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -12,10 +14,20 @@ ENV PYTHONPATH "${PYTHONPATH}:${APP_DIR}"
...
-    && python3.6 -m pip install -r ${APP_DIR}/requirements.txt
+    && python3.6 -m pip install -U pip \
+    && python3.6 -m pip install -r ${APP_DIR}/requirements.txt \
+    && git clone -b master https://${GITHUB_ACCESS_TOKEN}@github.com/DataTron-io/datatron_common/ \
+    && cd ${APP_DIR}/datatron_common/discovery \
+    && python3.6 -m pip wheel . \
+    && python3.6 -m pip install ${APP_DIR}/datatron_common/discovery/ \
+    && rm -rf ${APP_DIR}/datatron_common/ \
+    && cd ${APP_DIR}/app \
+    && git clone -b sankalp-metrics https://${GITHUB_ACCESS_TOKEN}@github.com/DataTron-io/governor/ \
```
Comment (dhruvrrp) [inline Dockerfile:30]: The branch should be master

**Review by dhruvrrp — COMMENTED** [review]
Left a comment

**Pull request merged**